no-downtime-upgrade/src/main/java/org/keycloak/playground/nodowntimeupgrade/base/model/ObjectModel_V3.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.playground.nodowntimeupgrade.base.model;
+
+/**
+ * Object model of a version {@link ModelVersion#VERSION_3}, one version older than the newest version.
+ * The newest version is supposed to be backward compatible with this version.
+ * @author hmlnarik
+ */
+public interface ObjectModel_V3 extends HasId<Integer> {
```

Comment:
Will Integer be default type for an ID of various objects in the new storage? Or is it used just for illustration purposes?

Reply:
It is here for illustration purposes, `Integer` here is one of the options available in the future Keycloak store. Databases would use some sort of binary ID, REST calls could use `String`s etc.

Reply:
Thanks for clarifying